src/components/SwitchToggle.js

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import styled from 'styled-components/macro';
+import PropTypes from 'prop-types';
+
+const Checkbox = styled.input.attrs({ type: 'checkbox' })`
+  border: 0;
+  clip: rect(0 0 0 0);
+  height: 1px;
+  margin: -1px;
+  overflow: hidden;
+  padding: 0;
+  position: absolute;
+  width: 1px;
+`;
+
+const Label = styled.label`
+  align-items: center;
+  display: flex;
+  min-height: 2rem;
+  padding-left: 3.5rem;
+  position: relative;
+  user-select: none;
+  &::before {
+    background-color: ${theme => theme.input};
+    border: 1px solid
+      ${({ checked, theme }) => (checked ? theme.border : '#ece9f1')};
+    border-radius: 100rem;
+    box-shadow: ${({ theme }) => theme.boxShadowSmall};
+    content: '';
+    display: block;
+    height: 1.75rem;
+    left: 0;
+    position: absolute;
+    top: 0.125rem;
+    transition: border-color ${({ theme }) => theme.transitionEase};
+    width: 3rem;
+  }
+  &::after {
+    background-color: ${({ checked, theme }) =>
+      checked ? theme.borderActive : theme.border};
+    border-radius: 50%;
+    content: '';
+    height: 1.5rem;
+    left: 0.125rem;
+    position: absolute;
+    top: 0.25rem;
+    transform: translateX(${props => (props.checked ? '1.25rem' : 0)});
+    transition: background-color ${({ theme }) => theme.transitionEase},
+      transform ${({ theme }) => theme.transitionEase};
```

Comment:
Didn't you want to add your easing, `transitionEaseOut` with the previous cubic-bezier? 😄